solr/contrib/ltr/src/test/org/apache/solr/ltr/ranking/TestReRankingPipeline.java

```diff
@@ -68,8 +71,11 @@ private IndexSearcher getSearcher(IndexReader r) {
       String field) {
     final List<Feature> features = new ArrayList<>();
     for (final int i : featureIds) {
-      final FieldValueFeature f = new FieldValueFeature("f" + i);
-      f.setField(field);
+      final Map<String,Object> params = new HashMap<String,Object>();
+      params.put("field", field);
+      final Feature f = Feature.getInstance(solrResourceLoader,
+          FieldValueFeature.class.getCanonicalName(),
+          "f" + i, i, params);
```

Comment:
There was no `f.setId(i)` to start with and so strictly speaking this is a change in behaviour (passing 0 would preserve existing code behaviour leaving multiple Feature objects with the same id).